.github/workflows/release.yml

```diff
@@ -0,0 +1,127 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Huawei Inc.
+#
+
+name: release
+on:
+  workflow_dispatch:
+    inputs:
+      ReleaseType:
+        type: choice
+        description: Select the version to released
+        options:
+          - Major Version
+          - Minor Version
+          - Patch Version
+
+env:
+  BOT_USER_NAME: eclipse-xpanse-bot
+  BOT_EMAIL_ID: xpanse-bot@eclipse.org
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  pre-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```

Comment:
we must checkout with a special token here - token: ${{ secrets.BOT_GITHUB_TOKEN }}
only this token has special permissions to write versions directly to 'main' branch.